kimchi/src/verifier.rs

```diff
@@ -762,28 +763,30 @@ pub fn verify<G, EFqSponge, EFrSponge>(
     group_map: &G::Map,
     verifier_index: &VerifierIndex<G>,
     proof: &ProverProof<G>,
+    public_input: &Vec<G::ScalarField>,
 ) -> Result<()>
 where
     G: KimchiCurve,
     G::BaseField: PrimeField,
     EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
     EFrSponge: FrSponge<G::ScalarField>,
 {
-    let proofs = vec![(verifier_index, proof)];
-    batch_verify::<G, EFqSponge, EFrSponge>(group_map, &proofs)
+    let proofs = vec![(proof, public_input)];
+    batch_verify::<G, EFqSponge, EFrSponge>(group_map, verifier_index, &proofs)
 }
 
 /// This function verifies the batch of zk-proofs
-///     proofs: vector of Plonk proofs
-///     index: `VerifierIndex`
+///     verifier_index: `VerifierIndex`
+///     proofs: vector of tuples (Plonk proof, public input)
 ///     RETURN: verification status
 ///
 /// # Errors
 ///
 /// Will give error if `srs` of `proof` is invalid or `verify` process fails.
 pub fn batch_verify<G, EFqSponge, EFrSponge>(
     group_map: &G::Map,
-    proofs: &[(&VerifierIndex<G>, &ProverProof<G>)],
+    verifier_index: &VerifierIndex<G>,
```

Comment:
I assume (perhaps out of my zero knowledge of the subject), that the same verifier_index would be used when performing batch verification.

Please let me know if that assumption is wrong!

Reply:
no actually here we expect to be passed a list of tuples (verifier_index, proof) so each proof can be associated with a different index (as long as they share the same SRS length

Reply:
(so now we should expect the public input to be another element in the tuple)

Reply:
Linter doesn't like the longer tuple, so introduced a struct. Seems to have made the code a little cleaner, actually.